dashboard/app/models/user.rb

```diff
@@ -605,10 +605,43 @@ def user_level_locked?(script_level, level)
     user_level.nil? || user_level.locked?(script_level.stage)
   end
 
+  # Returns the level in this script that the user most recently made progress on
+  def last_progressed_level(script)
+    user_levels.
+      where(script_id: script.id).
+      sort_by(&:updated_at).
+      last
+  end
+
+  # Returns the next script_level for the next progression level in the given
+  # script that hasn't yet been passed, starting its search at the last level we submitted
```

Comment:
Cool. I think we care a little about getting unexpected results (because we're running a migration for example), but not a lot. I think the alternative would be to store this "last progress level" somewhere else, which would mean an extra write every time we made progress, which seems less palatable.